packages/reporter/src/reporters/gitlab/builders/CodeQualityReportBuilder.ts

```diff
@@ -0,0 +1,91 @@
+import type {
+  CodeQualityReport,
+  CodeQualityIssue,
+  CodeQualitySeverity
+} from '../types';
+import type { Issue } from '@sectester/scan';
+import crypto from 'node:crypto';
+
+/**
+ * Builder for creating GitLab Code Quality reports from SecTester scan issues.
+ * Converts multiple issues into the GitLab Code Quality format.
+ */
+export class CodeQualityReportBuilder {
+  constructor(
+    private readonly issues: Issue[],
+    private readonly testFilePath: string
+  ) {}
+
+  /**
+   * Builds a GitLab Code Quality report from the provided issues.
+   *
+   * @returns An array of Code Quality issues in GitLab format
+   */
+  public build(): CodeQualityReport {
+    return this.issues.map(issue => this.convertIssueToCodeQualityIssue(issue));
+  }
+
+  /**
+   * Converts a SecTester issue to GitLab Code Quality issue format.
+   *
+   * @param issue - The SecTester issue to convert
+   * @returns The corresponding GitLab Code Quality issue
+   */
+  private convertIssueToCodeQualityIssue(issue: Issue): CodeQualityIssue {
+    const { originalRequest, name, severity } = issue;
+    const description = `${name} vulnerability found at ${originalRequest.method.toUpperCase()} ${originalRequest.url}`;
+
+    const fingerprint = this.createFingerprint(issue);
+
+    // Map sectester severity to GitLab severity
+    const gitlabSeverity = this.mapSeverity(severity);
+
+    return {
+      description,
+      check_name: name,
+      fingerprint,
+      severity: gitlabSeverity,
+      raw_details: JSON.stringify(issue, null, 2),
+      location: {
+        path: this.testFilePath,
+        lines: {
+          begin: 1,
+          end: 1
+        }
+      }
+    };
+  }
+
+  /**
+   * Creates a unique fingerprint for an issue based on its characteristics.
+   *
+   * @param issue - The issue to create a fingerprint for
+   * @returns A unique MD5 hash string
+   */
+  private createFingerprint(issue: Issue): string {
+    const content = `${issue.name}-${issue.originalRequest.method}-${issue.originalRequest.url}`;
```

Comment:
Use `entrypointId` as the more reliable approach

```suggestion
    const content = `${issue.name}-${issue.entrypointId}`;
```

Reply:
but we don't have `entrypointId` in the Issue. https://github.com/NeuraLegion/sectester-js/blob/feat/gitlab-code-quality-reporter/packages/scan/src/models/Issue.ts#L32

Reply:
we have https://app.brightsec.com/api/v1/docs/#/Issues/issuesGetScanIssues:~:text=%22entryPointId%22%3A%20%22upmVm5iPkddvzY6RisT7Cr%22%2C

Reply:
Why not use `issue.id` instead?

Reply:
It's always unique for each new scan, and won't work for deduplication. You could use `projectIssueId` instead, but `name` + `entrypointID` is preferable since it's more predictable for our logic